scene/main/node.cpp

```diff
@@ -71,6 +71,8 @@ void Node::_notification(int p_notification) {
 
 		} break;
 		case NOTIFICATION_ENTER_TREE: {
+			ERR_FAIL_COND(!get_viewport());
+			ERR_FAIL_COND(!get_tree());
```

Comment:
It also seems weird that any of those two can happen, `NOTIFICATION_ENTER_TREE` is emitted when the node enters the tree, which means that there should be a tree (and thus a root, so a viewport).

Reply:
I am suprised too. But if I don't check this way, the editor crashes. I think thats very rare condition.